comments-bundle/src/EventListener/DataContainer/CommentsConditionalFieldsListener.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CommentsBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+use Contao\CoreBundle\DependencyInjection\Attribute\AsHook;
+use Contao\System;
+
+#[AsHook('loadDataContainer')]
+class CommentsConditionalFieldsListener
+{
+    public function __invoke(string $table): void
+    {
+        if (!isset($GLOBALS['TL_DCA'][$table])) {
+            return;
+        }
+
+        switch ($table) {
+            case 'tl_module':
+                self::applyModuleFields();
+                break;
+
+            case 'tl_news_archive':
+            case 'tl_calendar':
+            case 'tl_faq_category':
+                self::applyParentFields($table);
+                break;
+
+            case 'tl_news':
+            case 'tl_calendar_events':
+            case 'tl_faq':
+                self::applyChildrenFields($table);
+                break;
+        }
+    }
+
+    private function applyModuleFields(): void
+    {
+        $bundles = System::getContainer()->getParameter('kernel.bundles');
```

Comment:
I don't think we should use `System::getContainer()` in regular services. Why not inject `%kernel.bundles%`?